services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -59,6 +59,16 @@ export const ChoiceListField = ({
 
     // update DOM choices checked status
     choices.forEach((choice: FieldChoice) => {
+      // if a choice is not selected and there are children, clear out any saved data
+      if (!choice.checked && choice.children) {
```

Comment:
Note: This works great as long as we only have checkedChildren. If we eventually have uncheckedChildren, we'll have to modify. Just a note for future us.